packages/hub/cli/db/transform-prisma-schema.ts

```diff
@@ -0,0 +1,161 @@
+      continue;
+    }
+
+    // We don't need to change anything if we aren't in a model or enum body
+    if (!currentModelName) {
+      fixedText.push(line);
+      continue;
+    }
+
+    // Add the @@map to the table name for the model
+    if (!hasAddedModelMap && (line.match(/\s+@@map/) || line === '}')) {
+      if (line === '}') {
+        fixedText.push('');
+        fixedText.push(`  @@map("${currentModelName}")`);
+      } else {
+        // Don’t add if the transformed name has already been added
+        const mappingMatch = line.match(/@@map\((\w+)\)/);
+
+        if (mappingMatch && mappingMatch[1] != currentModelName) {
+          fixedText.push(`  @@map("${currentModelName}")`);
+        }
+      }
+      hasAddedModelMap = true;
+    }
+
+    // Renames field and applies a @map to the field name if it is snake case
+    // Adds an s to the field name if the type is an array relation
+    const fieldMatch = line.match(/\s\s(\w+)\s+(\w+)(\[\])?/);
+    let fixedLine = line;
+    if (fieldMatch) {
+      const [, currentFieldName, currentFieldType, isArrayType] = fieldMatch;
+
+      let fixedFieldName = snakeToCamel(currentFieldName);
+      if (isArrayType && !pluralize.isPlural(fixedFieldName)) {
+        fixedFieldName = pluralize.plural(fixedFieldName);
+      }
+
+      fixedLine = fixedLine.replace(currentFieldName, fixedFieldName);
+
+      // Add map if we needed to convert the field name and the field is not a relational type
+      // If it's relational, the field type will be a non-primitive, hence the isPrimitiveType check
+      if (currentFieldName.includes('_') && isPrimitiveType(currentFieldType)) {
+        fixedLine = `${fixedLine} @map("${currentFieldName}")`;
+      }
+    }
+
+    // Capitalizes model names in field types
+    const fieldTypeMatch = fixedLine.match(/\s\s\w+\s+(\w+)/);
+    if (fieldTypeMatch) {
+      const currentFieldType = fieldTypeMatch[1];
+
+      const fieldTypeIndex = fieldTypeMatch[0].lastIndexOf(currentFieldType);
+      const fixedFieldType = pluralize.singular(snakeToPascal(currentFieldType));
+      const startOfLine = fixedLine.substr(0, fieldTypeIndex);
+      const restOfLine = fixedLine.substr(fieldTypeIndex + currentFieldType.length);
+      fixedLine = `${startOfLine}${fixedFieldType}${restOfLine}`;
+    }
+
+    // Changes `fields: [relation_id]` in @relation to camel case
+    const relationFieldsMatch = fixedLine.match(/fields:\s\[([\w,\s]+)\]/);
+    if (relationFieldsMatch) {
+      const fields = relationFieldsMatch[1];
+      fixedLine = fixedLine.replace(fields, fixFieldsArrayString(fields));
+    }
+
+    // Changes fields listed in @@index or @@unique to camel case
+    const indexUniqueFieldsMatch = fixedLine.match(/@@\w+\(\[([\w,\s]+)\]/);
+    if (indexUniqueFieldsMatch) {
+      const fields = indexUniqueFieldsMatch[1];
+      fixedLine = fixedLine.replace(fields, fixFieldsArrayString(fields));
+    }
+
+    fixedText.push(fixedLine);
+  }
+
```

Comment:
We can use `getDMMF` from `@prisma/internals` to assert that the data model matches what we expect.